framework/src/modules/nft/endpoint.ts

```diff
@@ -136,65 +136,75 @@ export class NFTEndpoint extends BaseEndpoint {
 		};
 	}
 
-	public async getCollectionIDs(
+	public async getSupportedCollectionIDs(
 		context: ModuleEndpointContext,
-	): Promise<{ collectionIDs: string[] }> {
+	): Promise<{ collectionIDs: string[] } | undefined> {
 		const { params } = context;
 
-		validator.validate<{ chainID: string }>(getCollectionIDsRequestSchema, params);
+		validator.validate<{ chainID: string }>(getSupportedCollectionIDsRequestSchema, params);
 
 		const chainID = Buffer.from(params.chainID, 'hex');
+		const nftID = Buffer.concat([chainID, Buffer.alloc(12)]);
 
-		const supportedNFTsStore = this.stores.get(SupportedNFTsStore);
-
-		const chainExists = await supportedNFTsStore.has(context.getImmutableMethodContext(), chainID);
+		const isNFTSupported = await this._nftMethod.isNFTSupported(
+			context.getImmutableMethodContext(),
+			nftID,
+		);
 
-		if (!chainExists) {
-			return { collectionIDs: [] };
+		if (!isNFTSupported) {
+			return undefined;
 		}
 
+		const supportedNFTsStore = this.stores.get(SupportedNFTsStore);
 		const supportedNFTsData = await supportedNFTsStore.get(
 			context.getImmutableMethodContext(),
 			chainID,
 		);
 
-		return {
-			collectionIDs: supportedNFTsData.supportedCollectionIDArray.map(collection =>
-				collection.collectionID.toString('hex'),
-			),
-		};
+		if (supportedNFTsData.supportedCollectionIDArray.length) {
+			return {
+				collectionIDs: supportedNFTsData.supportedCollectionIDArray.map(collection =>
+					collection.collectionID.toString('hex'),
+				),
+			};
+		}
+
+		return undefined;
```

Comment:
```suggestion
		return { collectionIDs: [] }; 
```

This return perhaps would execute if `supportedNFTsData.supportedCollectionIDArray` has of length 0, which highlights that all collections are valid, as LIP suggests, so an empty array of coolectionIDs is a valid case.

Reply:
The condition seemingly does point to the state when all collections are supported, `supportedCollectionIDArray` of length 0, and returning undefined might convey incorrect info.